pkg/asset/installconfig/powervs/validation.go

```diff
@@ -171,43 +171,53 @@ func ValidateCustomVPCSetup(client API, ic *types.InstallConfig) error {
 	return allErrs.ToAggregate()
 }
 
-func findVPCInRegion(client API, name string, region string, path *field.Path) field.ErrorList {
+func findVPCInRegion(client API, vpcNameOrID string, region string, path *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 
-	if name == "" {
+	if vpcNameOrID == "" {
 		return allErrs
 	}
 
-	vpcs, err := client.GetVPCs(context.TODO(), region)
+	ctx, cancel := context.WithTimeout(context.TODO(), 5*time.Minute)
+	defer cancel()
+
+	vpcs, err := client.GetVPCs(ctx, region)
 	if err != nil {
 		return append(allErrs, field.InternalError(path.Child("vpcRegion"), err))
 	}
 
-	found := false
 	for _, vpc := range vpcs {
-		if *vpc.Name == name {
-			found = true
-			break
+		if *vpc.Name == vpcNameOrID {
+			return allErrs
+		} else if *vpc.ID == vpcNameOrID {
+			return allErrs
 		}
 	}
-	if !found {
-		allErrs = append(allErrs, field.NotFound(path.Child("vpcName"), name))
-	}
+
+	allErrs = append(allErrs, field.NotFound(path.Child("vpcName"), vpcNameOrID))
 
 	return allErrs
 }
 
-func findSubnetInVPC(client API, subnets []string, region string, name string, path *field.Path) field.ErrorList {
+func findSubnetInVPC(client API, subnets []string, region string, vpcNameOrID string, path *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 
 	if len(subnets) == 0 {
 		return allErrs
 	}
 
-	subnet, err := client.GetSubnetByName(context.TODO(), subnets[0], region)
+	ctx, cancel := context.WithTimeout(context.TODO(), 5*time.Minute)
+	defer cancel()
+
+	subnet, err := client.GetSubnetByName(ctx, subnets[0], region)
 	if err != nil {
 		allErrs = append(allErrs, field.InternalError(path.Child("vpcSubnets"), err))
-	} else if *subnet.VPC.Name != name {
+	} else {
+		if *subnet.VPC.Name == vpcNameOrID {
```

Comment:
Similar to comment https://github.com/openshift/installer/pull/9818/files#r2196060940